### PR TITLE
Static schema support for Terraform Test and Mock files

### PR DESCRIPTION
**Pull request by dbanck**
This PR adds static schema support for Terraform Test and Mock files. This includes completions for Test specific block types, descriptions on hover for these blocks and the same for Mock files.

### CODE REVIEW

File: internal/schema/tests/1.6/variables_block.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+						schema.AttrNameStep{},
+					},
+					ScopeId:     refscope.VariableScope,
+					AsExprType:  true,
+					AsReference: true,
+				},
+				Constraint: schema.AnyExpression{OfType: cty.DynamicPseudoType},
+			},
+			Attributes: map[string]*schema.AttributeSchema{
+				"foo": {
```
Comment (jpogran) [inline internal/schema/tests/1.6/variables_block.go:33]: Is this supposed to be foo?

**Review by jpogran — CHANGES_REQUESTED** [review]
Great job! Love the documentation sourced from all the different places you had to look. One question about an attribute name, and then I think we intended to target main not pre-release

File: internal/schema/tests/1.6/variables_block.go (hunk 1 above)
Comment (ansgarm) [reply internal/schema/tests/1.6/variables_block.go:33]: I thought this should serve as an example – the AnyAttribute still allows to set any other value. But if we want to be correct, we should probably remove this 👍

Comment (ansgarm) [reply internal/schema/tests/1.6/variables_block.go]: removed!

**Review by jpogran — APPROVED** [review]
🏆

**Pull request merged**